server/routes/bookedtoday/arrivals/searchforexisting/changeDateOfBirthController.ts

```diff
@@ -1,14 +1,22 @@
 import type { RequestHandler } from 'express'
-import type { ExpectedArrivalsService } from '../../../../services'
+import { State } from './state'
 
 export default class ChangeDateOfBirthController {
-  public constructor(private readonly expectedArrivalsService: ExpectedArrivalsService) {}
-
   public showChangeDateOfBirth(): RequestHandler {
     return async (req, res) => {
-      const { id } = req.params
-      const data = await this.expectedArrivalsService.getArrival(id)
+      const data = State.searchDetails.get(req)
       res.render('pages/bookedtoday/arrivals/searchForExistingRecord/changeDateOfBirth.njk', { data })
     }
   }
+
+  public changeDateOfBirth(): RequestHandler {
+    return async (req, res) => {
+      const { id } = req.params
+      const { day, month, year } = req.body
+      const data = State.searchDetails.get(req)
+      // TODO build and validate
```

Comment:
Still ned to add validation, also want to add an `update` mechanism rather than manually get and overwrite everywhere